api/tests/test_teledeclaration.py

```diff
@@ -369,3 +370,101 @@ def test_complete_diagnostic(self):
         self.assertIn("value_fruits_et_legumes_france", json_teledeclaration)
         self.assertIn("value_fruits_et_legumes_short_distribution", json_teledeclaration)
         self.assertIn("value_fruits_et_legumes_local", json_teledeclaration)
+
+    @authenticate
+    def test_create_many_teledeclarations(self):
+        """
+        Given a list of diagnostic ids that are ready to TD, TD those diagnostics and return list of successful TD ids
+        """
+        canteen_1 = CanteenFactory.create()
+        canteen_2 = CanteenFactory.create()
+        for canteen in [canteen_1, canteen_2]:
+            canteen.managers.add(authenticate.user)
+
+        d_1 = DiagnosticFactory.create(canteen=canteen_1, value_total_ht=1000)
+        d_2 = DiagnosticFactory.create(canteen=canteen_2, value_total_ht=2000)
+        diagnostic_ids = [d_1.id, d_2.id]
+
+        response = self.client.post(
+            reverse("teledeclaration_create"), {"diagnosticIds": diagnostic_ids}, format="json"
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        body = response.json()
+        td_ids = body["teledeclarationIds"]
+        td = Teledeclaration.objects.get(id=td_ids[0])
```

Comment:
L'ordre des réponses n'est souvent pas garanti, ce qui fait des fails aléatoires dans les tests. On pourrait avoir un `assertIn` pour vérifier que les deux IDs sont à l'intérieur de `body["teledeclarationIds"]` avant de vérifier les données dans la BD.

Reply:
Les tests n'ont pas marché pour toi ? Dans le code, je fais l'itération de la liste fournie, et ajout les succès dans le même ordre, c'est pour ça que je pensais que je peux attendre un ordre particulier

Reply:
https://github.com/betagouv/ma-cantine/pull/2071/commits/af1978e1a7cec15c568e4eab6f5c6d1d163a9b06